src/ESports.ts

```diff
@@ -18,22 +18,172 @@ interface ESportsLeagueSchedule {
     teamB: string;
 }
 
+interface PickemLeaderboardEntry {
+    vsUserId: number;
+}
+
+interface PickemLeaderboard {
+    entries: PickemLeaderboardEntry[];
+}
+
+interface PickemTeam {
+    shortName: string;
+    name: string;
+    logoUrl: string;
+    wins: number;
+    losses: number;
+}
+
+interface PickemGroup {
+    name: string;
+    userPoints: number;
+    teams: PickemTeam[];
+}
+
+interface PickemUser {
+    summonerName: string;
+    id: number;
+}
+
+interface PickemGroupPick {
+    user: PickemUser;
+    groups: PickemGroup[];
+}
+
 export default class ESportsAPI {
     private bot: Discord.Client;
     private settings: SharedSettings;
 
     private schedule: Map<string, Map<string, ESportsLeagueSchedule[]>> = new Map();
 
+    private currentList: PickemGroupPick[];
+
     constructor(bot: Discord.Client, settings: SharedSettings) {
         this.bot = bot;
         this.settings = settings;
 
         bot.on("ready", async () => {
             await this.loadData();
             this.postInfo();
+            this.updateMemberList();
         });
     }
 
+    public async updateMemberList() {
+        this.currentList = await this.getMembersWithName();
+
+        setTimeout(this.updateMemberList.bind(this), 1000 * 60 * 60);
+    }
+
+    public async getCorrectPickem(): Promise<PickemGroupPick> {
+        const anyPick = await this.getPickem(4643536);
+        const best: PickemGroupPick = { user: { summonerName: "The Correct Choice", id: -1 }, groups: [] };
+
+        for (const group of anyPick.groups) {
+            best.groups.push({ name: group.name, teams: group.teams.sort((a, b) => b.wins - a.wins), userPoints: -1 });
+        }
+
+        return best;
+    }
+
+    public async getMembersWithName(): Promise<PickemGroupPick[]> {
+        const leaderboard = (await this.getLeaderboard()).entries;
+        const returnList: PickemGroupPick[] = [];
+
+        for (const entry of leaderboard) {
+            const pickem = await this.getPickem(entry.vsUserId);
+            returnList.push(pickem);
+        }
+
+        return returnList;
+    }
+
+    public async getLeaderboard(): Promise<PickemLeaderboard> {
+        const data = await fetch("https://pickem.euw.lolesports.com/en-GB/api/get_vs_lists/series/5/user/4643536");
+        return (await data.json())[0];
+    }
+
+    public async getPickem(id: number): Promise<PickemGroupPick> {
+        const url = "https://pickem.euw.lolesports.com/en-GB/api/get_group_picks/series/5/user/";
```

Comment:
hardcoded values, can you make this configurable

Reply:
You want the entire url to be configurable?